File: backend/src/main/java/com/yatmk/app/persistence/dto/BookDTO.java
package com.yatmk.app.persistence.dto;

import javax.validation.constraints.NotNull;

import org.springframework.web.multipart.MultipartFile;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BookDTO {

  private String author;

  private String country;

  private String language;

  private String link;

  private Long pages;

  private String title;

  private Long year;

  @NotNull
  private MultipartFile coverFile;

  @NotNull
  private MultipartFile contentFile;
}

File: back-end-nest/src/persistence/dto/book.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsString, IsNotEmpty, IsNumber, } from 'class-validator';

export class BookDTO {



  @IsString()
  @IsNotEmpty()
  @ApiProperty()
  author: string;

  @IsString()
  @IsNotEmpty()
  @ApiProperty()
  country: string;

  @IsString()
  @IsNotEmpty()
  @ApiProperty()
  language: string;

  @IsNumber()
  @IsNotEmpty()
  @ApiProperty()
  pages: number;

  @IsString()
  @IsNotEmpty()
  @ApiProperty()
  title: string;

  @IsNumber()
  @IsNotEmpty()
  @ApiProperty()
  year: number;

}